### ISSUE TITLE
Feyree EV charger shows temperature x10

### ISSUE THREAD

**Issue by tonysprenk** [make-all/tuya-local#1795]
This form is for reporting bugs.  If you have encountered a bug, you should
generally have some log messages or device diagnostics to show something is
going wrong.  If your problem is more of a question or a request for help,
then [Discussions](https://github.com/make-all/tuya-local/discussions) is
probably a more suitable venue.

**Describe the bug**
The temperature should be 34C not 340C

**To Reproduce**
Steps to reproduce the behavior including the device the issue was observed with:

**Expected behavior**

**Additional context**

If the bug involves a device, then please include device diagnostics from
Settings / Devices & Services / Tuya Local (your device) / "1 device"

Please check if there are messages from Tuya Local in the Home
Assistant log (Settings / System / Logs), and include those too.

**Comment by make-all** [comment]
Please attach device diagnostics.

**Comment by tonysprenk** [comment]
```
{
  "result": {
    "properties": [
      {
        "code": "work_state",
        "custom_name": "",
        "dp_id": 3,
        "time": 1707411104137,
        "value": "charger_free"
      },
      {
        "code": "fault",
        "custom_name": "",
        "dp_id": 10,
        "time": 1711178084664,
        "value": 0
      },
      {
        "code": "alarm_set_1",
        "custom_name": "",
        "dp_id": 11,
        "time": 1707411104137
      },
      {
        "code": "alarm_set_2",
        "custom_name": "",
        "dp_id": 12,
        "time": 1707411104137
      },
      {
        "code": "work_mode",
        "custom_name": "",
        "dp_id": 14,
        "time": 1707430791523,
        "value": "charge_schedule"
      },
      {
        "code": "balance_energy",
        "custom_name": "",
        "dp_id": 15,
        "time": 1707411104137,
        "value": 0
      },
      {
        "code": "clear_energy",
        "custom_name": "",
        "dp_id": 16,
        "time": 1707430770570,
        "value": false
      },
      {
        "code": "switch",
        "custom_name": "",
        "dp_id": 18,
        "time": 1707431403985,
        "value": false
      },
      {
        "code": "system_version",
        "custom_name": "",
        "dp_id": 23,
        "time": 1707411104137,
        "value": ""
      },
      {
        "code": "charge_energy_once",
        "custom_name": "",
        "dp_id": 25,
        "time": 1707411104137,
        "value": 1
      },
      {
        "code": "online_state",
        "custom_name": "",
        "dp_id": 27,
        "time": 1707411104137,
        "value": "online"
      },
      {
        "code": "DeviceState",
        "custom_name": "",
        "dp_id": 101,
        "time": 1712207775727,
        "value": "no_connect"
      },
      {
        "code": "Set32A",
        "custom_name": "",
        "dp_id": 102,
        "time": 1712142050166,
        "value": 32
      },
      {
        "code": "Set32Or16",
        "custom_name": "",
        "dp_id": 103,
        "time": 1711178084803,
        "value": true
      },
      {
        "code": "SetDelayTime",
        "custom_name": "",
        "dp_id": 104,
        "time": 1712141190397,
        "value": 0
      },
      {
        "code": "IDVerificationSet",
        "custom_name": "",
        "dp_id": 105,
        "time": 1707431384168,
        "value": false
      },
      {
        "code": "DeviceKwh",
        "custom_name": "",
        "dp_id": 106,
        "time": 1712207760504,
        "value": 0
      },
      {
        "code": "InputVoltage",
        "custom_name": "",
        "dp_id": 107,
        "time": 1712234449410,
        "value": 231
      },
      {
        "code": "OutCurrent",
        "custom_name": "",
        "dp_id": 108,
        "time": 1712207776311,
        "value": 0
      },
      {
        "code": "DeviceKw",
        "custom_name": "",
        "dp_id": 109,
        "time": 1712207774757,
        "value": 0
      },
      {
        "code": "DeviceTemp",
        "custom_name": "",
        "dp_id": 110,
        "time": 1712234566427,
        "value": 360
      },
      {
        "code": "Set16A",
        "custom_name": "",
        "dp_id": 111,
        "time": 1709764609683,
        "value": 10
      },
      {
        "code": "SwipRfid",
        "custom_name": "",
        "dp_id": 112,
        "time": 1712207775756,
        "value": false
      },
      {
        "code": "ctime",
        "custom_name": "",
        "dp_id": 113,
        "time": 1712207758617,
        "value": 0
      },
      {
        "code": "SetDefineTime",
        "custom_name": "",
        "dp_id": 114,
        "time": 1711178084988,
        "value": 0
      }
    ]
  },
  "success": true,
  "t": 1712234567062,
  "tid": "d6dc5d03f28011eeb283562879c6ad8c"
}

```

**Comment by tonysprenk** [comment]
This is how it is shown in home assistant. Seems like something is going wrong with the scaling even though it is set to “10” in the json for this device. 
![IMG_0790](https://github.com/make-all/tuya-local/assets/38426377/5263bb0c-0afb-4b1c-bb34-bce38c80d825)

**Comment by tonysprenk** [comment]
Yes sorry about that these are the diags:

```
  "integration_manifest": {
    "domain": "tuya_local",
    "name": "Tuya Local",
    "codeowners": [
      "@make-all"
    ],
    "config_flow": true,
    "dependencies": [],
    "documentation": "https://github.com/make-all/tuya-local",
    "integration_type": "device",
    "iot_class": "local_push",
    "issue_tracker": "https://github.com/make-all/tuya-local/issues",
    "requirements": [
      "tinytuya==1.13.2"
    ],
    "version": "2024.3.2",
    "is_built_in": false
  },
  "data": {
    "name": "EV charger tuya local",
    "type": "feyree_ev_charger",
    "device_id": "**REDACTED**",
    "device_cid": "",
    "local_key": "**REDACTED**",
    "host": "**REDACTED**",
    "protocol_version": 3.3,
    "tinytuya_version": "1.13.2",
    "api_version_set": 3.3,
    "api_version_used": 3.3,
    "api_working": true,
    "status": {},
    "cached_state": {
      "updated_at": 1712235504.5439544,
      "10": 0,
      "101": "no_connect",
      "102": 32,
      "103": true,
      "104": 0,
      "106": 0,
      "107": 234,
      "108": 0,
      "109": 0,
      "110": 361,
      "112": false,
      "113": 0,
      "114": 0
    },
    "pending_state": {},
    "connected": true,
    "force_dps": [],
    "home_assistant": {
      "name": "EV charger tuya local",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": "h",
          "state": {
            "entity_id": "number.ev_charger_tuya_local_charge_delay",
            "state": "0",
            "attributes": {
              "min": 0.0,
              "max": 15.0,
              "step": 1.0,
              "mode": "auto",
              "unit_of_measurement": "h",
              "icon": "mdi:car-clock",
              "friendly_name": "EV charger tuya local Charge delay"
            },
            "last_changed": "2024-04-03T15:00:11.878370+00:00",
            "last_updated": "2024-04-03T15:00:11.878370+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": "h",
          "state": {
            "entity_id": "number.ev_charger_tuya_local_charging_duration",
            "state": "0",
            "attributes": {
              "min": 0.0,
              "max": 15.0,
              "step": 1.0,
              "mode": "auto",
              "unit_of_measurement": "h",
              "icon": "mdi:car-clock",
              "friendly_name": "EV charger tuya local Charging duration"
            },
            "last_changed": "2024-04-03T15:00:11.878739+00:00",
            "last_updated": "2024-04-03T15:00:11.878739+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": "enum",
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "sensor.ev_charger_tuya_local_status",
            "state": "Disconnected",
            "attributes": {
              "options": [
                "Disconnected",
                "Connected",
                "Charging",
                "Waiting for authorization",
                "Charged",
                "Delaying charge",
                "Error"
              ],
              "fault_code": 0,
              "device_class": "enum",
              "icon": "mdi:ev-station",
              "friendly_name": "EV charger tuya local Status"
            },
            "last_changed": "2024-04-04T05:16:15.726081+00:00",
            "last_updated": "2024-04-04T05:16:15.726081+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": "energy",
          "icon": null,
          "unit_of_measurement": "kWh",
          "state": {
            "entity_id": "sensor.ev_charger_tuya_local_energy",
            "state": "0.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "kWh",
              "device_class": "energy",
              "friendly_name": "EV charger tuya local Energy"
            },
            "last_changed": "2024-04-04T05:16:00.505252+00:00",
            "last_updated": "2024-04-04T05:16:00.505252+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "diagnostic",
          "device_class": null,
          "original_device_class": "voltage",
          "icon": null,
          "unit_of_measurement": "V",
          "state": {
            "entity_id": "sensor.ev_charger_tuya_local_voltage",
            "state": "234",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "V",
              "device_class": "voltage",
              "friendly_name": "EV charger tuya local Voltage"
            },
            "last_changed": "2024-04-04T12:58:24.546578+00:00",
            "last_updated": "2024-04-04T12:58:24.546578+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": "current",
          "icon": null,
          "unit_of_measurement": "A",
          "state": {
            "entity_id": "sensor.ev_charger_tuya_local_current",
            "state": "0.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "A",
              "device_class": "current",
              "friendly_name": "EV charger tuya local Current"
            },
            "last_changed": "2024-04-04T05:16:16.335946+00:00",
            "last_updated": "2024-04-04T05:16:16.335946+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": "power",
          "icon": null,
          "unit_of_measurement": "kW",
          "state": {
            "entity_id": "sensor.ev_charger_tuya_local_power",
            "state": "0.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "kW",
              "device_class": "power",
              "friendly_name": "EV charger tuya local Power"
            },
            "last_changed": "2024-04-04T05:16:14.760875+00:00",
            "last_updated": "2024-04-04T05:16:14.760875+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "diagnostic",
          "device_class": null,
          "original_device_class": "temperature",
          "icon": null,
          "unit_of_measurement": "\u00b0C",
          "state": {
            "entity_id": "sensor.ev_charger_tuya_local_temperature",
            "state": "361",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "\u00b0C",
              "device_class": "temperature",
              "friendly_name": "EV charger tuya local Temperature"
            },
            "last_changed": "2024-04-04T12:57:29.039341+00:00",
            "last_updated": "2024-04-04T12:57:29.039341+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "diagnostic",
          "device_class": null,
          "original_device_class": "duration",
          "icon": null,
          "unit_of_measurement": "h",
          "state": {
            "entity_id": "sensor.ev_charger_tuya_local_time_remaining",
            "state": "0.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "h",
              "device_class": "duration",
              "friendly_name": "EV charger tuya local Time remaining"
            },
            "last_changed": "2024-04-04T05:15:58.630913+00:00",
            "last_updated": "2024-04-04T05:15:58.630913+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": "restart",
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "button.ev_charger_tuya_local_clear_energy",
            "state": "unknown",
            "attributes": {
              "device_class": "restart",
              "friendly_name": "EV charger tuya local Clear energy"
            },
            "last_changed": "2024-04-03T15:00:11.879014+00:00",
            "last_updated": "2024-04-03T15:00:11.879014+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "button.ev_charger_tuya_local_toggle_charging",
            "state": "2024-04-03T10:46:01.289898+00:00",
            "attributes": {
              "friendly_name": "EV charger tuya local Toggle charging"
            },
            "last_changed": "2024-04-03T15:00:11.879275+00:00",
            "last_updated": "2024-04-03T15:00:11.879275+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": null,
          "icon": "n",
          "unit_of_measurement": "A",
          "state": {
            "entity_id": "number.ev_charger_tuya_local_set_charge_current",
            "state": "32",
            "attributes": {
              "min": 8.0,
              "max": 32.0,
              "step": 1.0,
              "mode": "auto",
              "max_current": 32,
              "unit_of_measurement": "A",
              "icon": "n",
              "friendly_name": "EV charger tuya local Set charge current"
            },
            "last_changed": "2024-04-03T15:00:11.877918+00:00",
            "last_updated": "2024-04-03T16:37:12.817496+00:00"
          }
        }
      ]
    }
  }
}
```

**Comment by make-all** [comment]
This is already fixed by PR #1730 (not yet in a release)
